Port to libwbmqtt1 0.2 and refactoring

PeteK12 opened the pull request:
- Добавил очистку старых контролов
- Сделал вывод ошибок в /meta/error
- Сделал рефакторинг
- Добавил комментариев
- Перевёл на systemd
- Восстановил тесты

KraPete committed:
Port to libwbmqtt1 0.2 and refactoring

KraPete committed:
Exit with error after failed test

evgeny-boger commented on debian/wb-homa-w1.service:
```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Kernel 1-Wire MQTT driver for WB-HomA
+Wants=wb-hwconf-magager.service wb-modules.service
```
опечатка эта же

PeteK12 replied:
Да, блин. Поправил

evgeny-boger commented on onewire_driver.cpp:
```diff
@@ -1,285 +1,144 @@
 
-    if (Worker->joinable()) {
-        Worker->join();
+    DeleteControls(part.OnlyFirst, Device, tx, InfoLogger);
```
кажется, это то же самое, что ты недавно реализовывал в libwbmqtt

PeteK12 replied:
Я делал удаление того, что мы не создавали в этом сеансе. Тут же удаление контролов, которые мы до этого сами создавали. Т.е. пропал каталог термометра в sysfs, мы удаляем контрол.

evgeny-boger commented on main.cpp:
```diff
@@ -1,128 +1,189 @@
-            w1_driver.Clear();
-        });
+        {
+            TThreadedPeriodicalRunner r(
```
не очень понял, а зачем это делать в отдельном потоке да ещё и через особую штуку? Чем это здесь хуже, чем ```while(1) { sleep(Interval)}```

?

PeteK12 replied:
Я просто разбил старый код на 2 класса. Там тоже в отдельном потоке крутилось. Sleep не очень, т.к. замедленная реакция на внешние сигналы.  В старом коде интервал задавался в мс, а sleep на секунду делался. Переделал на condition variable, теперь поток спит и просыпается в нужное время в мс. 2 класса удобнее тестировать, не высовывая детали реализации наружу.

KraPete committed:
Fix typo

KraPete committed:
Rename to wb-mqtt-w1

PeteK12 commented:
Переименовал в wb-mqtt-w1

evgeny-boger committed:
remove g++ build dependency to allow cross-build

as per https://www.debian.org/doc/manuals/maint-guide/dreq.ru.html,
"Some packages like gcc and make which are required by the build-essential package are implied"

evgeny-boger committed:
Merge pull request #7 from wirenboard/feature/gxx-deps

remove g++ build dependency to allow cross-build

KraPete committed:
Service alias to wb-homa-w1 is added

KraPete committed:
Cross compilation support

KraPete committed:
Disable lto for tests to avoid compiler bug

PeteK12 committed:
Merge pull request #9 from wirenboard/feature/cross-compile-support

Cross compilation support

evgeny-boger committed:
Merge pull request #8 from wirenboard/feature/service-alias

Service alias to wb-homa-w1 is added
